From 86d9f77cbc1657ffff2a5ba9cf7ff6715ce9c9b5 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Wed, 27 Apr 2022 02:57:46 -0400
Subject: [PATCH] add-eip93-crypto

---
 Kconfig  | 2 +-
 Makefile | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Kconfig b/Kconfig
index 1f6308c..c65a055 100644
--- a/drivers/crypto/Kconfig
+++ b/drivers/crypto/Kconfig
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0-only
-
+source "drivers/crypto/mtk-eip93/Kconfig"
 menuconfig CRYPTO_HW
 	bool "Hardware crypto devices"
 	default y
diff --git a/Makefile b/Makefile
index afc4753..9b7cd66 100644
--- a/drivers/crypto/Makefile
+++ b/drivers/crypto/Makefile
@@ -1,4 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
+obj-$(CONFIG_CRYPTO_DEV_EIP93) += mtk-eip93/
 obj-$(CONFIG_CRYPTO_DEV_ALLWINNER) += allwinner/
 obj-$(CONFIG_CRYPTO_DEV_ATMEL_AES) += atmel-aes.o
 obj-$(CONFIG_CRYPTO_DEV_ATMEL_SHA) += atmel-sha.o
-- 
2.17.1

