From 58b6b805786718b537fa290d1a693cc1e90c319d Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Sun, 17 Apr 2022 01:13:16 +0900
Subject: [PATCH] revert Use CRIOGET

---
 crypto/engine/eng_devcrypto.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/crypto/engine/eng_devcrypto.c b/crypto/engine/eng_devcrypto.c
index 84a3b7dbe..877a6774c 100644
--- a/crypto/engine/eng_devcrypto.c
+++ b/crypto/engine/eng_devcrypto.c
@@ -758,9 +758,8 @@ static int devcrypto_unload(ENGINE *e)
 void engine_load_devcrypto_int()
 {
     ENGINE *e = NULL;
-    int fd;
 
-    if ((fd = open("/dev/crypto", O_RDWR, 0)) < 0) {
+    if ((cfd = open("/dev/crypto", O_RDWR, 0)) < 0) {
 #ifndef ENGINE_DEVCRYPTO_DEBUG
         if (errno != ENOENT)
 #endif
@@ -768,18 +767,6 @@ void engine_load_devcrypto_int()
         return;
     }
 
-#ifdef CRIOGET
-    if (ioctl(fd, CRIOGET, &cfd) < 0) {
-        fprintf(stderr, "Could not create crypto fd: %s\n", strerror(errno));
-        close(fd);
-        cfd = -1;
-        return;
-    }
-    close(fd);
-#else
-    cfd = fd;
-#endif
-
     if ((e = ENGINE_new()) == NULL
         || !ENGINE_set_destroy_function(e, devcrypto_unload)) {
         ENGINE_free(e);
-- 
2.17.1

