From f5c0226abcfa38afce2592c0aba839b3b2b6cb72 Mon Sep 17 00:00:00 2001
From: W_Y_CPP <383152993@qq.com>
Date: Fri, 22 Apr 2022 00:28:44 -0400
Subject: [PATCH] fix build kernel 5.4

---
 net/batman-adv/netlink.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/net/batman-adv/netlink.c b/net/batman-adv/netlink.c
index 00875e1d8..cd28d7bd0 100644
--- a/net/batman-adv/netlink.c
+++ b/net/batman-adv/netlink.c
@@ -6,7 +6,7 @@
 
 #include "netlink.h"
 #include "main.h"
-
+#include <linux/version.h>
 #include <linux/atomic.h>
 #include <linux/bitops.h>
 #include <linux/bug.h>
@@ -1356,8 +1356,11 @@ static void batadv_post_doit(const struct genl_ops *ops, struct sk_buff *skb,
 		dev_put(bat_priv->soft_iface);
 	}
 }
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0) 
 static const struct genl_small_ops batadv_netlink_ops[] = {
+#else
+static const struct genl_ops batadv_netlink_ops[] = {
+#endif
 	{
 		.cmd = BATADV_CMD_GET_MESH,
 		.validate = GENL_DONT_VALIDATE_STRICT | GENL_DONT_VALIDATE_DUMP,
@@ -1491,8 +1494,13 @@ struct genl_family batadv_netlink_family __ro_after_init = {
 	.pre_doit = batadv_pre_doit,
 	.post_doit = batadv_post_doit,
 	.module = THIS_MODULE,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 10, 0) 
 	.small_ops = batadv_netlink_ops,
 	.n_small_ops = ARRAY_SIZE(batadv_netlink_ops),
+#else
+	.ops = batadv_netlink_ops,
+	.n_ops = ARRAY_SIZE(batadv_netlink_ops),
+#endif
 	.mcgrps = batadv_netlink_mcgrps,
 	.n_mcgrps = ARRAY_SIZE(batadv_netlink_mcgrps),
 };
-- 
2.17.1

